src/template.py

```
@@ -0,0 +1,420 @@
+                    break
+            except subprocess.CalledProcessError:
+                if waitclose:
+                    break
+                else:
+                    time.sleep(1)
+            if timeout is not None and timeout > 0:
+                if time.time() - starttime > timeout:
+                    raise Exception('timeout')
+
+    def docker_build(self):
+        result = subprocess.run(
```

Ruff (0.12.2)

`subprocess` call with `shell=True` identified, security issue

(S602)